autogluon/task/text_classification/dataset.py

```diff
@@ -119,28 +122,167 @@ def get_dataset(self, segment='train'):
         sampler, _ = get_split_samplers(dataset, split_ratio=0.2)
         return SampledDataset(dataset, sampler)
 
-class TSVClassificationTask(AbstractGlueTask):
-    def __init__(self, *args, **kwargs): # passthrough arguments to TSVDataset
-        # (filename, field_separator=nlp.data.Splitter(','), num_discard_samples=1, field_indices=[2,1])
-        self.args = args
-        self.kwargs = kwargs
-        is_pair = False
-        class_labels = ['0', '1']
-        metric = CompositeEvalMetric()
-        metric.add(F1())
-        metric.add(Accuracy())
-        super(TSVClassificationTask, self).__init__(class_labels, metric, is_pair)
-        dataset = nlp.data.TSVDataset(*self.args, **self.kwargs)
-        # do the split
-        train_sampler, val_sampler = get_split_samplers(dataset, split_ratio=0.8)
-        self.trainset = SampledDataset(dataset, train_sampler)
-        self.valset = SampledDataset(dataset, val_sampler)
+class CustomTSVClassificationTask(AbstractGlueTask):
```

Comment:
why it is a subclass of `AbstractGlueTask`, even that it has same the data format as the GLUE dataset, but semantically it's a not a GLUE task.

Reply:
Agreed & to add to this, can we call it something like "TSVDataset" instead?  
Naming is important here since this is user-facing and will appear in the docs.
I especially don't think the word "Custom" needs to appear, whether or not "Task" or "Classification" should be there is up to you, but it clashes with the fact that the usual workflow starts with:
```
import TextClassification as task 
```
(I think "Task" remains ok for the other GLUE dataset objects since those datasets are all associated with a very particular task beyond generic TextClassification).

Reply:
Please find my reply: https://github.com/awslabs/autogluon/pull/363#discussion_r394675221
Task is more proper for NLP pipeline. Dataset is not comprehensive enough